settings/src/lib.rs

```diff
@@ -224,6 +243,33 @@ impl RitaCommonSettings<RitaSettingsStruct> for Arc<RwLock<RitaSettingsStruct>>
             self.get_network().wg_public_key.clone(),
         )
     }
+
+    fn get_stats<'ret, 'me: 'ret>(
+        &'me self,
+    ) -> RwLockReadGuardRef<'ret, RitaSettingsStruct, StatServerSettings> {
+        RwLockReadGuardRef::new(self.read().unwrap()).map(|g| match g.stat_server {
+            Some(ref stat_server) => stat_server,
+            None => panic!("exit client not set but needed"),
+        })
+    }
+
+    fn init_stats(&self, stat_server: StatServerSettings) {
+        self.write().unwrap().stat_server = Some(stat_server)
+    }
+
+    fn set_stats<'ret, 'me: 'ret>(
+        &'me self,
+    ) -> RwLockWriteGuardRefMut<'ret, RitaSettingsStruct, StatServerSettings> {
+        RwLockWriteGuardRefMut::new(self.write().unwrap()).map_mut(|g| match g.stat_server {
+            Some(ref mut stat_server) => stat_server,
+            None => panic!("exit client not set but needed"),
```

Comment:
This is a pattern which I use a lot in the settings code, basically, whenever I have an optional field in the main settings struct, I do this so you can do .optinal_stuff() instead of .optional_stuff().clone().unwrap() or .optional_stuff().clone()? (clone is needed as unwrap requires owning the struct)

Reply:
I was thinking more about the `panic!`. Isn't it a silly reason to crash the whole of rita? (that's where it runs in, no?)